test/Guardian.t.sol

```diff
@@ -0,0 +1,367 @@
+        vm.startPrank(user);
+        lendingPool.borrowUnguarded(100);
+        vm.stopPrank();
+
+        // Then: the total borrow is updated
+        assertEq(lendingPool.totalBorrow(), 100);
+    }
+
+    function testRevert_borrowGuarded_paused() public {
+        emit log_named_uint("pauset", lendingPool.pauseTimestamp());
+        emit log_named_uint("times", block.timestamp);
+        // Given: the lending pool is paused
+        vm.startPrank(guardian);
+        lendingPool.pause();
+        vm.stopPrank();
+
+        // Given: only borrow left paused
+        vm.startPrank(owner);
+        lendingPool.unPause(false, false, true, false, false);
+        vm.stopPrank();
+
+        // When: a user tries to supply
+        vm.startPrank(user);
+        lendingPool.depositGuarded(100);
+        vm.stopPrank();
+
+        // Then: the total supply is updated
+        assertEq(lendingPool.totalSupply(), 100);
+
+        // When: user tries to borrow, which is paused
+        vm.expectRevert("Guardian: borrow paused");
+        vm.startPrank(user);
+        lendingPool.borrowGuarded(100);
+        vm.stopPrank();
+
+        // Then: the total borrow is not updated
+        assertEq(lendingPool.totalBorrow(), 0);
+
+        // Revert: the lending pool is unpaused
+        vm.startPrank(owner);
+        lendingPool.unPause(false, false, false, false, false);
+        vm.stopPrank();
+    }
+
+    function testSuccess_unPause_ownerCanUnPauseDuring30Days(uint256 timePassedAfterPause) public {
+        vm.assume(timePassedAfterPause <= 30 days);
+
+        // Given: the lending pool is paused
+        vm.startPrank(guardian);
+        lendingPool.pause();
+        vm.stopPrank();
+
+        // Given: Sometime passed after the pause
+        vm.warp(block.timestamp + timePassedAfterPause);
+
+        // When: the owner unPauses the supply
+        vm.startPrank(owner);
+        lendingPool.unPause(true, true, true, false, true);
+        vm.stopPrank();
+
+        // Then: the user can supply
+        vm.startPrank(user);
+        lendingPool.depositGuarded(100);
+        vm.stopPrank();
+
+        // Then: the total supply is updated
+        assertEq(lendingPool.totalSupply(), 100);
+    }
+
+    function testSuccess_unPause_onlyUnpausePossible(uint256 timePassedAfterPause) public {
+        vm.assume(timePassedAfterPause <= 30 days);
+
+        // Given: the lending pool is paused
+        vm.startPrank(guardian);
+        lendingPool.pause();
+        vm.stopPrank();
+
+        // Given: Sometime passed after the pause
+        vm.warp(block.timestamp + timePassedAfterPause);
+
+        // When: the owner unPauses the supply
+        vm.startPrank(owner);
+        lendingPool.unPause(true, true, true, false, true);
+        vm.stopPrank();
+
+        // When: the owner attempts the pause the supply from the unPause
+        vm.startPrank(owner);
+        lendingPool.unPause(true, true, true, true, true);
```

Comment:
It should not be able to be paused when calling 'unPause'. This function call do nothing

Reply:
This test exactly checks that. To be sure that unPause call does nothing